aip/general/0129.md

```diff
@@ -0,0 +1,71 @@
+---
+id: 129
+state: draft
+created: 2023-06-21
+placement:
+  category: resource-design
+  order: 65
+---
+
+# Server-modified Values and Defaults
```

Comment:
Thanks for writing this!

I think it would be fantastic to frame fields as having a "single owner" principle, before going into client-server and permadiff as an example of problems from violating that principle.

The principle is that each field of a resource may be determined by one-and-only-one software component. The obvious problem being when any field is owned by two components the resource will oscillate between states.

Because GCP is a client-server architecture, being output_only has a clear meaning. And for cases where there's a single client the most apparent oscillation is between client and server. 

But in other cases the principle stands up --- in k8s controllers are also clients, so status fields are writable, but only one component-client is properly entitled to modify any given field. (field manager metadata helps track that.)

And even in client-server cases the principle extends to multiple clients. If there are any read-write fields which a non-Terraform software component will be driving, then the Terraform client must be told explicitly to ignore_changes for converging state.

Reply:
I did a large rework and used the single owner framing